src/iconUploader.js

```
@@ -0,0 +1,29 @@
+const fs = require('fs')
+const path = require('path')
+const axios = require('axios')
+const FormData = require('form-data')
+
+const PROJECT_ROOT = path.resolve(process.cwd(), '..')
+
+const uploadIcons = async (icons, pluginId, folderName) => {
+  for (const [key, value] of Object.entries(icons)) {
+    const nodeType = key
+    const nodePath = value
+    const nodeFilePath = path.join(PROJECT_ROOT + `/${folderName}/` + nodePath)
+    const formData = new FormData()
+    formData.append('icon', fs.createReadStream(nodeFilePath), {
+      contentType: 'image/png',
+      filename: path.basename(nodeFilePath)
+    })
+    try {
+      await axios.post(process.env.CRAFTER_API_URL + `/plugins/icons/${nodeType}/${pluginId.id}`, formData, {
+        headers: { ...formData.getHeaders(), 'User-Agent': 'cpc/1.0' }
+      })
+      console.log(`${nodeType} icon uploaded successfully`)
+    } catch (error) {
+      console.error('failed to upload icons', error)
```

Lets throw the error to caller with appropriate error message